test/adaptive_load/metrics_plugin_test.cc

```diff
@@ -0,0 +1,233 @@
+
+TEST_P(NighthawkStatsEmulatedMetricsPluginTestFixture, ComputesCorrectMetric) {
+  NighthawkStatsEmulatedMetricsPlugin plugin =
+      NighthawkStatsEmulatedMetricsPlugin(MakeSimpleNighthawkOutput({
+          /*concurrency=*/"auto",
+          /*requests_per_second=*/1024,
+          /*actual_duration_seconds=*/10,
+          /*upstream_rq_total=*/2560,
+          /*response_count_2xx=*/320,
+          /*min_ns=*/400,
+          /*mean_ns=*/500,
+          /*max_ns=*/600,
+          /*pstdev_ns=*/11,
+      }));
+  const std::string& metric_name = std::get<0>(GetParam());
+  const double value = std::get<1>(GetParam());
+  EXPECT_EQ(plugin.GetMetricByName(metric_name).value(), value);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    NighthawkStatsEmulatedMetricsPluginValuesTests, NighthawkStatsEmulatedMetricsPluginTestFixture,
+    ::testing::Values(std::make_tuple<std::string, double>("attempted-rps", 1024),
+                      std::make_tuple<std::string, double>("achieved-rps", 256),
+                      std::make_tuple<std::string, double>("send-rate", 0.25),
+                      std::make_tuple<std::string, double>("success-rate", 0.125),
+                      std::make_tuple<std::string, double>("latency-ns-min", 400.0),
+                      std::make_tuple<std::string, double>("latency-ns-mean", 500.0),
+                      std::make_tuple<std::string, double>("latency-ns-max", 600.0),
+                      std::make_tuple<std::string, double>("latency-ns-mean-plus-1stdev", 511.0),
+                      std::make_tuple<std::string, double>("latency-ns-mean-plus-2stdev", 522.0),
+                      std::make_tuple<std::string, double>("latency-ns-mean-plus-3stdev", 533.0),
+                      std::make_tuple<std::string, double>("latency-ns-pstdev", 11.0)));
+
+TEST(NighthawkStatsEmulatedMetricsPluginTest, ReturnsErrorIfGlobalResultMissing) {
+  nighthawk::client::Output empty_output;
+  NighthawkStatsEmulatedMetricsPlugin plugin = NighthawkStatsEmulatedMetricsPlugin(empty_output);
+  EXPECT_THAT(plugin.GetMetricByName("x").status().message(),
+              testing::HasSubstr("'global' result not found"));
+}
+
+TEST(NighthawkStatsEmulatedMetricsPluginTest, ReturnsErrorIf2xxMissing) {
+  nighthawk::client::Output output = MakeSimpleNighthawkOutput({
+      /*concurrency=*/"auto",
+      /*requests_per_second=*/1024,
+      /*actual_duration_seconds=*/10,
+      /*upstream_rq_total=*/2560,
+      /*response_count_2xx=*/320,
+      /*min_ns=*/400,
+      /*mean_ns=*/500,
+      /*max_ns=*/600,
+      /*pstdev_ns=*/11,
+  });
+  output.mutable_results(0)->clear_counters();
+  NighthawkStatsEmulatedMetricsPlugin plugin = NighthawkStatsEmulatedMetricsPlugin(output);
+  EXPECT_THAT(plugin.GetMetricByName("x").status().message(),
+              testing::HasSubstr("'benchmark.total_2xx' not found"));
+}
+
+TEST(NighthawkStatsEmulatedMetricsPluginTest, ReturnsErrorIfUpstreamRqTotalMissing) {
```

Comment:
Could these two not be the same test with both expectations? Or is that bad practice?

Reply:
I split this into two tests because it's technically two behaviors. Theoretically a future rewrite could break one behavior but not the other.

> Not sure my comments were sensible.

Absolutely, thanks for the review!